billing/downloader.go

```diff
@@ -0,0 +1,147 @@
+package billing
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/cycloidio/raws"
+)
+
+type Downloader interface {
+	Download(dest string) (string, error)
+}
+
+type billingDownloader struct {
+	connector    *raws.Connector
+	s3Bucket     string
+	filename     string
+	fileFullPath string
+}
+
+func NewDownloader(s3Connector *raws.Connector, bucket, filename string) Downloader {
+	return &billingDownloader{
+		connector: s3Connector,
+		filename:  filename,
+		s3Bucket:  bucket,
+	}
+}
+
+func (d *billingDownloader) Download(dest string) (string, error) {
+	fullPath, err := d.getAndCreateOutputPath(dest)
+	if err != nil {
+		return "", fmt.Errorf("Error while identifying destination's path: %v", err)
+	}
+	d.fileFullPath = fullPath
+	fd, err := os.Create(d.fileFullPath)
+	defer fd.Close()
```

Comment:
The `defer fd.Close()` should go after checking the errors, because if there is any error the `fd` is nil and you will have a null pointer exception.